src/components/NewProjectFlow/flow/ProjectEntitySearch.tsx

```diff
@@ -0,0 +1,108 @@
+import { Grid, Box, useTheme } from '@mui/material';
+import { PillListItem, Textfield } from '@terraware/web-components';
+import { PillList } from '@terraware/web-components';
+import strings from 'src/strings';
+import { ProjectEntityFilters } from 'src/components/NewProjectFlow/flow/useProjectEntitySelection';
+import { useAppSelector } from 'src/redux/store';
+import { selectProjects } from 'src/redux/features/projects/projectsSelectors';
+import ProjectEntityFilter, { EntitySpecificFilterConfig } from './ProjectEntityFilter';
+
+interface ProjectEntitySearchProps {
+  searchValue: string;
+  onSearch: (value: string) => void;
+  entitySpecificFilterConfigs: EntitySpecificFilterConfig[];
+  filters: ProjectEntityFilters;
+  setFilters: (value: ProjectEntityFilters) => void;
+}
+
+export type PillListItemWithEmptyValue = PillListItem<string> & { emptyValue: unknown };
+
+export default function ProjectEntitySearch(props: ProjectEntitySearchProps): JSX.Element | null {
+  const { searchValue, onSearch, entitySpecificFilterConfigs, filters, setFilters } = props;
+
+  const theme = useTheme();
+
+  const projects = useAppSelector(selectProjects);
+
+  const [filterPillData, setFilterPillData] = useState<PillListItemWithEmptyValue[]>([]);
+
+  const onRemovePillList = useCallback(
+    (filterId: string) => {
+      const filter = filterPillData?.find((filterPillDatum) => filterPillDatum.id === filterId);
+      setFilters({ [filterId]: filter?.emptyValue || null });
+    },
+    [filterPillData, setFilters]
+  );
+
+  const projectEntityFilterConfig: EntitySpecificFilterConfig = useMemo(
+    () => ({
+      label: strings.PROJECT,
+      initialSelection: filters.projectIds || [],
+      filterKey: 'projectIds',
+      options: projects?.map((project) => project.id) || [],
+      renderOption: (projectId: string | number) =>
+        (projects || []).find((project) => project.id === projectId)?.name || '',
+      pillModifier: (_filters: ProjectEntityFilters): PillListItemWithEmptyValue[] => {
+        const projectIds = _filters.projectIds || [];
+        if (projectIds.length === 0) {
+          return [];
+        }
+
+        return [
+          {
+            id: 'projectIds',
+            label: strings.PROJECT,
+            value: projectIds
+              .map((projectId: number) => (projects || []).find((project) => project.id === projectId))
+              .map((project) => project?.name)
+              .join(','),
+            emptyValue: [],
+          },
+        ];
+      },
+    }),
+    [filters.projectIds, projects]
+  );
+
+  useEffect(() => {
+    const data: PillListItemWithEmptyValue[] = [
+      ...projectEntityFilterConfig.pillModifier(filters),
+      ...entitySpecificFilterConfigs.map((config) => config.pillModifier(filters)).flat(),
+    ];
+
+    setFilterPillData(data);
+  }, [entitySpecificFilterConfigs, filters, projectEntityFilterConfig]);
+
+  return (
+    <>
```

Comment:
Was wondering if this component can be leveraged.. https://github.com/terraware/terraware-web/blob/main/src/components/common/SearchFiltersWrapper.tsx .. looks like the use-case here in Projects is a tad more custom.

Reply:
Yea that does look like it could be used. I think there would be some work making it work with the rest of the flow and it looks like I would have to re-work the filter pills because they would be inline next to each filter, instead of at the end. Let me know if you think I should go that route

Reply:
Thanks for taking a look at that. What we have here in this PR is better organized for the use-case. Refactoring the generic component may work but could be a forced effort perhaps. My vote is to not go that route. Worth discussing I thought.